modules/simphera_base/k8s.tf

```diff
@@ -61,7 +61,7 @@ resource "azurerm_kubernetes_cluster" "aks" {
   }
 
   dynamic "api_server_access_profile" {
-    for_each = length(var.apiServerAuthorizedIpRanges) > 0 ? [1] : []
+    for_each = var.apiServerAuthorizedIpRanges == null ? [] : tolist([var.apiServerAuthorizedIpRanges])
```

Comment:
I don't understand this change. What is the idea behind it?

Reply:
So, I've done some tests, with multiple different inputs (eg, var set to null - default, some value and multiple values).
It did not compute correctly, so I've changed the approach and now it does (for same tests).

Reply:
The pattern `[1] : []` is a [hack to realize conditional dynamic blocks](https://reespozzi.medium.com/conditional-nested-configuration-blocks-for-terraform-resources-57d8ce372bad). What was the error message you got?

Reply:
I believe this intended semantics are:
apiServerAuthorizedIpRanges = null => no IPs are blocked
apiServerAuthorizedIpRanges = [] => all IPs are blocked
apiServerAuthorizedIpRanges = ["1.2.3.4/24"] => all IP ranges except 1.2.3.4/24 are blocked

Reply:
Have you tested your solution with two ranges, e.g. `apiServerAuthorizedIpRanges = ["1.2.3.4/24", "4.3.2.1/32"]`? I expect it to fail.
If it fails, can you try `for_each = var.apiServerAuthorizedIpRanges == null ? [] : [1]` ?

Reply:
I've done following tests (cluster create-destroy-create again):  
api_server_authorized_ip_ranges 
1. null -> configured as [], can access k8s cluster remotely  
2. "" -> not configured, Error: "The given value is not suitable for var.apiServerAuthorizedIpRanges declared at variables.tf:180,1-39: set of string required"  
3. [""] -> not configured, Error: "Error: api_server_access_profile.0.authorized_ip_ranges.0 must start with IPV4 address and/or slash, number of bits (0-32) as prefix. Example: 127.0.0.1/8. Got "" "  
4. single ip -> configured as [192.168.0.0/16], can not access k8s cluster remotely (IP dependant)  
5. multiple ips -> configured as ["0.0.0.0/0","10.0.0.0/8","192.168.0.0/16",], can access k8s cluster remotely (IP dependant)

Reply:
case 1 and 4 are not supposed to be equivalent, but that is not how I understand the code and I believe it is not what you mean.

My understanding is that if apiServerAuthorizedIpRanges has a length of n >1 that the dynamic block is overridden n times (with the same content). I guess `for_each = var.apiServerAuthorizedIpRanges == null ? [] : tolist([var.apiServerAuthorizedIpRanges])` should do the job, but so should `for_each = var.apiServerAuthorizedIpRanges == null ? [] : [1]`. If the items (i.e. the individual IP ranges) in the array would matter, we would see a `each.value` inside the loop.

Reply:
It is not what I mean but what is actually configured in cluster at the end. I think I covered all use-cases (ie test with both positive and negative inputs and observe the results). For me, it does not make sense that you can configure cluster, so that you are initially locked out of it and require additional post-install steps to configure it. Var behavior is exactly the same across both "solutions", there is nothing modifying the actual value.  

I am not saying this did not work before - it did, I deployed it before etc. What I am saying is that I am getting errors on my current test setup, WSL2 + versions listed, and I am fixing them one by one. So, we need to agree on supported platforms (Win only or?) and versions of tools used. Since this is just var passing, provider versions play no part here, but there could be slightly different behavior between TF versions.

Reply:
Are we sure that this change does not break execution on Windows?

Reply:
Tested cases 1. and 6., in that particular order. This covers everything, in my opinion, as others TCs are either variations or will produce errors.  
There was no cluster destroy, only update, since there's no possibility of value merge or overwrite. Same end result on Win setup as well, and no "hacks" necessary.